pkgs/data/misc/cacert/default.nix

```
@@ -1,123 +1,211 @@
 { lib
 , stdenv
 , fetchurl
-, nss
-, python3
-, blacklist ? [ ]
+, buildcatrust
+, blacklist ? []
+, extraCertificates ? []
 
-  # Used for tests only
+# Used for overriding certdata version (see update.sh)
+, nssOverride ? null
+
+# Used for tests only
 , runCommand
 , cacert
 , openssl
 }:
 
+# extraCertificates should be paths or strings.
+assert lib.length (lib.filter (f: !(builtins.isPath f || builtins.isString f)) extraCertificates) == 0;
```

We probably want to allow results of `runCommand` or `writeText` as well?

```suggestion
assert lib.length (lib.filter (f: !(builtins.isPath f || builtins.isString f || lib.isDerivation f)) extraCertificates) == 0;
```

---

I'm going to rip this out, since I have those anyway in the NixOS module and this is needlessly complicated.